hostmap.go

```diff
@@ -395,9 +403,15 @@ func (hm *HostMap) DeleteReverseIndex(index uint32) {
 	}
 }
 
-func (hm *HostMap) DeleteHostInfo(hostinfo *HostInfo) {
+// DeleteHostInfo will fully unlink the hostinfo and return true if it was the final hostinfo for this vpn ip
+func (hm *HostMap) DeleteHostInfo(hostinfo *HostInfo) bool {
 	// Delete the host itself, ensuring it's not modified anymore
 	hm.Lock()
+	final := true
+	if hostinfo.next != nil || hostinfo.prev != nil {
+		// If we have a previous or next hostinfo then we are not the last one for this vpn ip
+		final = false
+	}
```

Comment:
Just a nit, but maybe a little clearer?

```suggestion
        // If we have a previous or next hostinfo then we are not the last one for this vpn ip
	final := (hostinfo.next == nil && hostinfo.prev == nil)
```